gufe/solventcomponent.py

```diff
@@ -65,6 +66,10 @@ def concentration(self) -> unit.Quantity:
         """Concentration of ions in the solvent state"""
         return self._concentration
 
+    def formal_charge(self) -> int:
+        """Solvents don't have a formal charge defined so this returns nan"""
+        return math.nan
```

Comment:
mypy doesn't like this -- `math.nan` is a float, not an int

Reply:
`None` would make typing easier but math harder; OTOH, maybe we want to fail fast if user adds an unknown charge, rather than make them hunt down the source of a `nan` in their code?

Reply:
I would advocate for `None` as a "there is nothing here" - `NaN` feels like there is an indication of something existing but it's just wrong?

Reply:
I think I'm leaning more toward `None`, despite the backflips we get to do in handling in e.g. `sum`. Solves the `mypy` issue and agree with @dwhswenson and @IAlibay notes above.

Reply:
Ok went with None